logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt

```diff
@@ -2196,12 +2201,31 @@ class UserSessionScope internal constructor(
             kaliumLogger = userScopedLogger,
         )
 
+    private val analyticsRepository: AnalyticsRepository
+        get() = AnalyticsDataSource(
+            userDAO = userStorage.database.userDAO,
+            selfUserId = userId,
+            metadataDAO = userStorage.database.metadataDAO
+        )
+
     val getTeamUrlUseCase: GetTeamUrlUseCase
         get() = GetTeamUrlUseCase(
             userId,
             authenticationScope.serverConfigRepository,
         )
 
+    val getAnalyticsContactsData: GetAnalyticsContactsDataUseCase get() = GetAnalyticsContactsDataUseCaseImpl(
+        selfTeamIdProvider = selfTeamId,
+        slowSyncRepository = slowSyncRepository,
+        analyticsRepository = analyticsRepository,
+        userConfigRepository = userConfigRepository
+    )
+
+    val asyncUpdateContactsAmountsCache: AsyncUpdateContactsAmountsCacheUseCase get() = AsyncUpdateContactsAmountsCacheUseCase(
```

Comment:
Why expose this as a UseCase to the clients? It's offloading logic to clients.

`GetAnalyticsContactsData` could refresh its data source if needed when invoked, without having to delegate the app to do trigger it.

Reply:
i was thinking about this, my idea was to return the result and then trigger the update if needed, i guess i will invest some time to do so